EduVPN/Models/Crypto.swift

```diff
@@ -14,9 +14,44 @@ enum CryptoError: Error {
 }
 
 class Crypto {
-    private static let keyName = "disk_storage_key"
-
-    private static func makeAndStoreKey(name: String) throws -> SecKey {
+    
+    static var shared: Crypto {
+        return self.sharedInstance
+    }
+    
+    private static let sharedInstance = Crypto()
+    
+    private let keyName = "disk_storage_key"
+    
+    private let hasSecurityEnclave: Bool
+    private let algorithm: SecKeyAlgorithm
+    private let keySize: Int
+    private let keyType: CFString
+    
+    init() {
+        var attributes = [String: Any]()
+        attributes[kSecAttrKeyType as String] = kSecAttrKeyTypeEC
+        attributes[kSecAttrKeySizeInBits as String] = 256
+        
+        var error: Unmanaged<CFError>?
+        let _ = SecKeyCreateRandomKey(attributes as CFDictionary, &error)
+        
+        if let error = error?.takeRetainedValue(),
+            CFErrorGetCode(error) == errSecUnimplemented {
+            
+            self.hasSecurityEnclave = true
```

Comment:
This seems to be the other way around? Would the failure too create the key mean that there is no security enclave?

Reply:
I think you are right.
...

Reply:
The code attempts to create a key type related to the Secure Enclave.

So, NO error should mean there is a Secure Enclave.